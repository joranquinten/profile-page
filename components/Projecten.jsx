import React from 'react';
import styled from 'styled-components';
import renderHTML from 'react-render-html';

import { Card } from 'antd';

import { colors } from '../styled/themes/default';

import { projects } from '../data/nl';

const StyledProjecten = styled.div`
  .ant-card {
    border-color: ${colors.blackDarkened};
    background: ${colors.blackLightened};
    color: ${colors.white};
    margin: 0 0 2rem;

    strong {
      color: ${colors.accentColor};
    }

  .ant-card-head {
    border-color: ${colors.blackDarkened};
  }

  .ant-card-head-title {
    color: ${colors.tertiaryColor};
  }
`;

const Projecten = () => (
  <StyledProjecten>
    {projects.map(item => (
      <Card title={item.title} key={item.title}>
        {renderHTML(item.description)}
      </Card>
    ))}
  </StyledProjecten>
);

export default Projecten;
