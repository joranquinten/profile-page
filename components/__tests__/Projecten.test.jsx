import { shallow } from 'enzyme';

import Projecten from '../Projecten';

jest.mock('../../data/nl.js');

describe('<Projecten />', () => {
  it('should render without crashing', () => {
    expect(shallow(<Projecten />)).toMatchSnapshot();
  });
});
