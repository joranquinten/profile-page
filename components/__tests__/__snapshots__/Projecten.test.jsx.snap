// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projecten /> should render without crashing 1`] = `
<Projecten__StyledProjecten>
  <Card
    key="Project one"
    title="Project one"
  >
    <p
      key="0"
    >
      Project one description
    </p>
  </Card>
  <Card
    key="Project two"
    title="Project two"
  >
    <p
      key="0"
    >
      Project two description
    </p>
  </Card>
</Projecten__StyledProjecten>
`;
